Prose/ProseLib/Sources/ProseCoreTCA/ProseClient+Live.swift

```diff
@@ -0,0 +1,145 @@
+import Combine
+import ComposableArchitecture
+import Foundation
+@_implementationOnly import ProseCore
+@_implementationOnly import ProseCoreClientFFI
+import SharedModels
+
+private extension Account {
+    static let placeholder = Account(jid: "void@prose.org", status: .connected)
+}
+
+public extension ProseClient {
+    static func live(
+        date: @escaping () -> Date = Date.init,
+        uuid: @escaping () -> UUID = UUID.init
+    ) -> Self {
+        // Only one client/account for now.
+        var client: ProseCore.ProseClient?
+        let delegate = Delegate(date: date)
+
+        return ProseClient(
+            login: { jid, password in
+                delegate.account.value = .init(jid: jid, status: .connecting)
+                client = .init(delegate: delegate)
+
+                do {
+                    try client?.authenticate(jid: jid.rawValue, with: .password(password))
+                    try client?.connect()
+                } catch {
+                    return Effect(error: EquatableError(error))
+                }
+
+                return delegate.account
+                    .tryDrop { account in
+                        switch account.status {
+                        case .connecting:
+                            return true
+                        case .connected:
+                            return false
+                        case let .error(error):
+                            throw error
+                        }
+                    }
+                    .handleEvents(receiveOutput: { _ in
+                        try? client?.loadRoster()
+                    })
+                    .mapError(EquatableError.init)
+                    .map { _ in .none }
+                    .eraseToEffect()
+            },
+            logout: { _ in
+                Empty(completeImmediately: true).eraseToEffect()
+            },
+            roster: {
+                delegate.roster
+                    .map { roster in
+                        // Append our own JID to each group, so that we can chat with
+                        // ourselves (via a second IM).
```

Comment:
Why "via a second IM"? Couldn't we chat with ouselves as Slack allows to? I regularly us this feature. I hope we'll support this someday.

Reply:
That's not up to me. This change allowed me to send messages back and forth between Prose and Adium which is helpful during development.

Reply:
> That's not up to me.

I don't understand your point, I feel like we agree it's a good feature 🤔 I'm just saying it should not require another IM app, a single Prose window (or 2 for manual testing when we'll support multi-account) should be enough to send a message to yourself.

Reply:
But you're not testing the whole flow then. If you send a message to someone else the server doesn't send you a copy of your message. So our code has to add that message to the conversation in memory. If you use another IM you can be sure that messages are sent and received on either side.

Reply:
Yes yes I agree, I just wanted to talk about a specific feature 😉